src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx

```diff
@@ -113,6 +124,28 @@ class DatePicker extends React.Component {
         }
     };
 
+    async onChangeCalendarMonth(calendar_date) {
+        const trading_events = await getTradingEvents(calendar_date, this.props.underlying);
+        const holidays = [];
+        let weekends   = [];
+        trading_events.filter(events => {
```

Comment:
How come you are using `.filter` here? I think e.g. foreach is more appropriate since filter returns a new filtered array

Reply:
good catch, thanks :)